src/main/haskell/kore/test/Test/Kore.hs

```diff
@@ -237,78 +270,125 @@ domainValueGen _ x =
+impliesGen vars childGen x = binaryOperatorGen vars childGen x Implies
 
-inGen :: MetaOrObject level => Gen child -> level -> Gen (In level child)
-inGen childGen x = equalsInGen childGen x In
+inGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
+    -> level
+    -> Gen (In level child)
+inGen vars childGen x = equalsInGen vars childGen x In
 
-nextGen :: MetaOrObject level => Gen child -> level -> Gen (Next level child)
-nextGen childGen x = unaryOperatorGen childGen x Next
+nextGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
+    -> level
+    -> Gen (Next level child)
+nextGen vars childGen x = unaryOperatorGen vars childGen x Next
 
-notGen :: MetaOrObject level => Gen child -> level -> Gen (Not level child)
-notGen childGen x = unaryOperatorGen childGen x Not
+notGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
+    -> level
+    -> Gen (Not level child)
+notGen vars childGen x = unaryOperatorGen vars childGen x Not
 
-orGen :: MetaOrObject level => Gen child -> level -> Gen (Or level child)
-orGen childGen x = binaryOperatorGen childGen x Or
+orGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
+    -> level
+    -> Gen (Or level child)
+orGen vars childGen x = binaryOperatorGen vars childGen x Or
 
 rewritesGen
-    :: MetaOrObject level => Gen child -> level -> Gen (Rewrites level child)
-rewritesGen childGen x = binaryOperatorGen childGen x Rewrites
+    :: MetaOrObject level
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
+    -> level
+    -> Gen (Rewrites level child)
+rewritesGen vars childGen x = binaryOperatorGen vars childGen x Rewrites
 
-topGen :: MetaOrObject level => level -> Gen (Top level child)
-topGen x = topBottomGen x Top
+topGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> level
+    -> Gen (Top level child)
+topGen vars  x = topBottomGen vars x Top
 
 patternGen
     :: MetaOrObject level
-    => Gen child
+    => [Variable level]
+    -> ([Variable level] -> Gen child)
     -> level
     -> Gen (Pattern level Variable child)
-patternGen childGen x =
-    oneof
-        [ AndPattern <$> andGen childGen x
-        , ApplicationPattern <$> applicationGen childGen x
-        , BottomPattern <$> bottomGen x
-        , CeilPattern <$> ceilGen childGen x
-        , EqualsPattern <$> equalsGen childGen x
-        , ExistsPattern <$> existsGen childGen x
-        , FloorPattern <$> floorGen childGen x
-        , ForallPattern <$> forallGen childGen x
-        , IffPattern <$> iffGen childGen x
-        , ImpliesPattern <$> impliesGen childGen x
-        , InPattern <$> inGen childGen x
-        , NotPattern <$> notGen childGen x
-        , OrPattern <$> orGen childGen x
-        , TopPattern <$> topGen x
-        , VariablePattern <$> variableGen x
+patternGen vars childGen x =
+    frequency
```

Comment:
What is the reason for using `frequency` here?

Reply:
VariablePattern is too rare to obtain nontrivial test cases otherwise.

Reply:
At least, that was what I thought when I was inspecting it manually. Maybe it's silly.